### ISSUE TITLE
Remove Endpoints - letter-branding

### ISSUE THREAD

**Issue by EvanParish** [department-of-veterans-affairs/notification-api#1149]
## User Story - Business Need

In the effort to clean up our code we need to investigate and remove unused endpoints. This ticket is to take a look at the endpoints under `/letter-branding/` (app/letter_branding/letter_branding_rest.py) and remove the ones marked for removal. You can view the table of endpoints [here](https://github.com/department-of-veterans-affairs/vanotify-team/blob/master/Engineering/SPIKES/VA%20Notify%20Endpoints.md).

**This requires a methodical approach**. The letter branding endpoints utilize several dao methods, each of those should be removed and the associated references and models. There's a letter-brand schema that should be cleaned out. It should be ensured that no methods or variables created in letter_branding_rest.py are used anywhere else in our code (e.g. `letter_branding_blueprint`).

This **must** be a document-as-you-go process. Keep everything in one doc and pair-down as needed when publishing it to the team repo.

- [x] Ticket is understood, and QA has been contacted (if the ticket has a QA label).

### User Story(ies)
**As an** engineer
**I want** to clean up and reduce our codebase
**So that** it is easier to maintain and upgrade in the future.

### Additional Info and Resources
- [Endpoint Table](https://github.com/department-of-veterans-affairs/vanotify-team/blob/master/Engineering/SPIKES/VA%20Notify%20Endpoints.md)
- [Endpoint Spreadsheet](https://docs.google.com/spreadsheets/d/18AfqlEOWlrnigauO57jA0VP3RMHTaP4luknqBxrcbuY/edit#gid=115873237)

## Engineering Checklist

**Strike QA (Jake) will be needed on this ticket. Please keep this in mind and keep them looped in throughout the ticket in addition to Cris.**

- [x] Start a github md file to document this process and update as you go
- [x] Draft PR opened immediately & save deploying to Dev until the full changes are ready (use local testing until then)
- [x] Frequent commits to draft PR - **post in eng chat know when a commit is ready** for @k-macmillan @kalbfled @EvanParish to review
- [x] Touch base with QA at all forks in the road
- [x] When deploying this branch be sure that all developers understand that a migration is involved with your branch.
- [x] Investigate the endpoints starting with `/letter-branding` to make sure they aren't being used
- [x] Also look at the `/letters/returned` () endpoint since it's probably not worth having a separate ticket for that.
- [x] Remove tests associate with those endpoints
- [x] Remove the DB table and any columns in other tables that represent letter-branding. 
- [x] Remove the code associated with those endpoints (if possible). Make sure not to remove any code used by other parts of the application.

## Acceptance Criteria

- [ ] When the designated code and tests have been removed, the application retains full functionality. 
- [x] Process is thoroughly documented and put in the [team repo](https://github.com/department-of-veterans-affairs/vanotify-team).

## QA Considerations
- [ ] The QA Regression Suite passes
- [ ] The Cypress regression and any Portal testing needed passes

## Potential Dependencies
- Make sure the removal of these endpoints doesn't affect the work Strike is doing.

**Comment by mjones-oddball** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Notify-620d21369d810a00146ed9c8/issues/department-of-veterans-affairs/notification-api/1149?planning-poker) with Zenhub @EvanParish @justaskdavidb2 @k-macmillan @kalbfled @ldraney @cris-oddball @jakehova

**Comment by kalbfled** [comment]
```bash
$ grep -rnil -E "letter(_|-)branding" app/
app/__pycache__/models.cpython-310.pyc
app/__pycache__/__init__.cpython-310.pyc
app/letter_branding/letter_branding_schema.py
app/letter_branding/letter_branding_rest.py
app/commands.py
app/celery/letters_pdf_tasks.py
app/dao/services_dao.py
app/dao/letter_branding_dao.py
app/schemas.py
app/template/rest.py
app/__init__.py
app/service/rest.py
app/models.py
```

**Comment by cris-oddball** [comment]
This branch was deployed to Dev. 

A regression against the public endpoints passes (after creating a new QA branch to not check for properties now removed from the GET /services endpoints)
` 134 passed, 3 xfailed in 266.53s (0:04:26)`

**Comment by cris-oddball** [comment]
Now that running regressions in the cloud is fixed, it passes the regression against the private endpoints too
134 passed, 3 xfailed in 179.41s (0:02:59)

**Comment by nikolai-efimov** [comment]
https://github.com/department-of-veterans-affairs/vanotify-team/blob/master/Process/Coding/UnusedCodeRemoval.md

**Comment by nikolai-efimov** [comment]
waiting for a response from @Jake-Uhteg about testing this from his side. @cris-oddball suggested that we don't merge this until we here from Jake.

**Comment by cris-oddball** [comment]
Code is currently re-deployed to Dev and over to @Jake-Uhteg for UI testing.

**Comment by cris-oddball** [comment]
Jake confirmed the UI looks good in testing.

**Comment by cris-oddball** [comment]
Code approved and merged to master, overriding the certifi issue, since that is excluded in the branch for v3 POST and a ticket is upcoming to upgrade certifi.

Deploying to Perf for testing.

**Comment by cris-oddball** [comment]
Regression in Perf passes 134 passed, 3 xfailed in 144.11s (0:02:24)
Just going to do some poking around before closing.

**Comment by cris-oddball** [comment]
Not seeing any errors that could be related to this on Perf in the past day. Closing ticket.
